### PR TITLE
Parallelize deserialization of incoming BASP messages

### PR DESCRIPTION
**Pull request by Neverlord**
This is a followup to https://github.com/actor-framework/actor-framework/pull/821 with fixes for the issues that required to revert the changes via https://github.com/actor-framework/actor-framework/pull/828.

### CODE REVIEW

File: libcaf_io/src/basp_broker.cpp [hunk 1]
```diff
@@ -121,6 +121,7 @@ behavior basp_broker::make_behavior() {
     send(this, tick_atom::value, heartbeat_interval);
   }
   return {
+
```
Comment (josephnoir) [inline libcaf_io/src/basp_broker.cpp]: O'rly?

File: libcaf_io/src/basp_broker.cpp [hunk 2]
```diff
@@ -144,17 +145,17 @@ behavior basp_broker::make_behavior() {
...
         message_id mid, const message& msg) {
-      CAF_LOG_TRACE(CAF_ARG(src) << CAF_ARG(dest)
-                    << CAF_ARG(mid) << CAF_ARG(msg));
+      CAF_LOG_TRACE(CAF_ARG(src)
+                    << CAF_ARG(dest) << CAF_ARG(mid) << CAF_ARG(msg));
       if (!dest || system().node() == dest->node()) {
-        CAF_LOG_WARNING("cannot forward to invalid or local actor:"
-                        << CAF_ARG(dest));
+        CAF_LOG_WARNING(
+          "cannot forward to invalid or local actor:" << CAF_ARG(dest));
```
Comment (josephnoir) [inline libcaf_io/src/basp_broker.cpp]: I'm not sure about this one either. Doesn't really matter though.

File: libcaf_io/src/worker.cpp [hunk 3]
```diff
@@ -31,11 +31,11 @@ namespace basp {
 // -- constructors, destructors, and assignment operators ----------------------
 
 worker::worker(worker_hub& hub, message_queue& queue, proxy_registry& proxies)
-    : next_(nullptr),
-      hub_(&hub),
-      queue_(&queue),
-      proxies_(&proxies),
-      system_(&proxies.system()) {
+  : next_(nullptr),
```
Comment (josephnoir) [inline libcaf_io/src/worker.cpp]: Our [contributing](https://github.com/actor-framework/actor-framework/blob/master/CONTRIBUTING.md#breaking-statements) document says something else. There are a few other constructors that were formatted this way in this commit.

File: libcaf_io/src/basp_broker.cpp (hunk 1 above)
Comment (Neverlord) [reply libcaf_io/src/basp_broker.cpp]: This newline only kept `clang-format` from formatting the lambdas below aligned after the `{`. I did find a better solution though: `behavior{` is a tiny bit more verbose but it gets `clang-format` to correctly align the lambdas two-spaces from the left.

File: libcaf_io/src/basp_broker.cpp (hunk 2 above)
Comment (Neverlord) [reply libcaf_io/src/basp_broker.cpp]: I've broken up the string literal to get a nicer formatting.

**Pull request merged**